src/main/java/com/ibm/cloud/cloudant/kafka/mappers/SinkRecordToDocument.java

```diff
@@ -78,38 +81,37 @@ private Map<String, Object> convertStruct(Struct struct, Map<String, Object> out
         // iterate fields and add to map
         for (Field f : schema.fields()) {
             Object value = struct.get(f);
-            outMap.put(f.name(), getField(f.schema().type(), value));
+            outMap.put(f.name(), convertItemFromStruct(f.schema().type(), value));
         }
         return outMap;
     }
 
     // convert kafka map to map by adding key/values to passed in map, and returning it
     private Map<String, Object> convertMap(Map<?,?> inMap, Map<String, Object> outMap) {
 
+        // iterate over keys and add to map
         for (Object k : inMap.keySet()) {
             if (k instanceof String) {
                 Object v = inMap.get(k);
-                if (v instanceof Map) {
-                    outMap.put((String) k, convertMap((Map<?,?>) v, new HashMap<>()));
-                } else if (v instanceof Struct) {
-                    outMap.put((String) k, convertStruct((Struct) v, new HashMap<>()));
-                } else {
-                    // assume that JSON serialiser knows how to deal with it
-                    outMap.put((String) k, v);
-                }
+                outMap.put((String) k, convertItem(v));
             } else {
                 throw new IllegalArgumentException("unsupported type in map key " + k.getClass());
             }
```

Comment:
counterpart of the above suggestion
```suggestion
                outMap.put((String) k, convertItem(v));
            } else {
                throw new IllegalArgumentException("unsupported type in map key " + k.getClass());
            }
```